src/kv/MemDB.cc

```diff
@@ -397,7 +401,7 @@ void MemDB::MDBWholeSpaceIteratorImpl::fill_current()
 
 bool MemDB::MDBWholeSpaceIteratorImpl::valid()
 {
-  if (m_key_value.first.empty()) {
+  if (!(*is_valid) || m_key_value.first.empty()) {
```

Comment:
After keeping track of seqno, I think just restarting the iterator with m_key_value.first  as key should work. m_iter = m_btree_p->lower_bound(k);